crates/red_knot_workspace/tests/check.rs

```diff
@@ -1,48 +1,102 @@
-use std::fs;
 use std::path::PathBuf;
 
 use red_knot_python_semantic::{HasTy, SemanticModel};
 use red_knot_workspace::db::RootDatabase;
 use red_knot_workspace::workspace::WorkspaceMetadata;
-use ruff_db::files::{system_path_to_file, File};
+use ruff_db::files::{system_path_to_file, File, Files};
 use ruff_db::parsed::parsed_module;
-use ruff_db::system::{OsSystem, SystemPath, SystemPathBuf};
+use ruff_db::system::{SystemPath, SystemPathBuf, TestSystem};
 use ruff_python_ast::visitor::source_order;
 use ruff_python_ast::visitor::source_order::SourceOrderVisitor;
 use ruff_python_ast::{self as ast, Alias, Expr, Parameter, ParameterWithDefault, Stmt};
 
-fn setup_db(workspace_root: &SystemPath) -> anyhow::Result<RootDatabase> {
-    let system = OsSystem::new(workspace_root);
+fn setup_db(workspace_root: &SystemPath, system: TestSystem) -> anyhow::Result<RootDatabase> {
     let workspace = WorkspaceMetadata::from_path(workspace_root, &system, None)?;
     RootDatabase::new(workspace, system)
 }
 
-/// Test that all snippets in testcorpus can be checked without panic
+fn get_workspace_root() -> anyhow::Result<PathBuf> {
+    Ok(PathBuf::from(String::from_utf8(
+        std::process::Command::new("cargo")
+            .args(["locate-project", "--workspace", "--message-format", "plain"])
+            .output()?
+            .stdout,
+    )?)
```

Comment:
Nit: Use a `SystemPathBuf` here to "retain" the information that this is a valid UTF8 path
```suggestion
    Ok(SystemPathBuf::from(String::from_utf8(
        std::process::Command::new("cargo")
            .args(["locate-project", "--workspace", "--message-format", "plain"])
            .output()?
            .stdout,
    )?)
```